uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestStatusMatrixTest.php

```diff
@@ -73,28 +73,101 @@ public function testCanTransitionToForLeaveApproverReturnsFalseForAllNonPossible
   }
 
   public function testCanTransitionToForAdminForReturnsTrueAllPossibleTransitionStatuses() {
+    $adminID = 5;
+    $leaveContact = 2;
+    $this->registerCurrentLoggedInContactInSession($adminID);
+    $this->setPermissions(['administer leave and absences']);
+
+    $possibleTransitions = $this->allPossibleStatusTransitionForLeaveApprover();
+
+    foreach($possibleTransitions as $transition) {
+      $this->assertTrue($this->leaveRequestStatusMatrix->canTransitionTo($transition[0], $transition[1], $leaveContact));
+    }
+  }
+
+  public function testCanTransitionToForAdminReturnsFalseForAllNonPossibleTransitionStatuses() {
+    $adminID = 5;
+    $leaveContact = 2;
```

Comment:
I see the fabricator is used in tests below - why not here too?

`$leaveContact = ContactFabricator::fabricate();`